cloud/disk_manager/internal/pkg/dataplane/url/common/http_client.go

```diff
@@ -174,19 +203,16 @@ func (c *httpClient) head(
 		return nil, errors.NewRetriableError(err)
 	}
 
-	if resp.StatusCode == 403 {
-		return nil, NewSourceForbiddenError("http code %d", resp.StatusCode)
-	}
-
-	if resp.StatusCode >= 400 && resp.StatusCode <= 499 {
-		return nil, NewSourceNotFoundError("http code %d", resp.StatusCode)
+	err = checkFor4xxOr5xxStatusCode(resp.StatusCode)
+	if err != nil {
+		return nil, err
 	}
 
-	if resp.StatusCode >= 500 && resp.StatusCode <= 599 {
-		return nil, errors.NewRetriableErrorf("http code %d", resp.StatusCode)
+	if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
```

Comment:
why not do this check inside `checkHttpStatus` and return nil if status is 2xx and error in other cases

Reply:
good point

fixed